vispy/visuals/image.py

```diff
@@ -401,7 +401,8 @@ def _build_texture(self):
             # can eventually do this by simulating 32-bit float... maybe
             clim = self._clim
             if isinstance(clim, string_types) and clim == 'auto':
-                clim = np.min(data), np.max(data)
+                finite_data = data[np.isfinite(data)]
```

Comment:
@djhoese As I wrote above, it would be nice to issue a Warning if the source data contains NaN/INF, so the user has at least an information that the image might not be what he expects. Other wise I'm fully in favour of adding this.

Reply:
Is there a way to do that without an even bigger performance hit? You said that not warning about it and doing it this way will lead to misinterpretation of the data. I get that, but I'm on the fence as to whether this is that likely. Of course now that I've said that it is going to happen to me.

Reply:
Could looks like this :

```python
isfinite = np.isfinite(data)
if isfinite.any():
    from warnings import warn
    warn("Image contains non-finite values that have been ignored to compute colorbar limits")
    # maybe logger.warning() would be better?
finite_data = data[isfinite]
clim = np.min(finite_data), np.max(finite_data)
```

Reply:
One way to stay performant is just to penalize the people who need penalizing, e.g.:
```
min_ = np.nanmin(data)
max_ = np.nanmax(data)
if not np.isfinite(min) or not np.isfinite(max_):
    warn(...)
```
then people with only nan+finite pay no performance penalty. In this case I wouldn't even bother with the `warn` and just say in the docstring that if you have infinite values in your data, the number of passes for computing min/max will double

Reply:
That's a nice solution, however here you need to perform `nanmin` and `nanmax`, which by definition ignore NaN and are just looking for inf. Maybe simply `np.min` and `np.max` could work? I've heard that nan NumPy operators are slower

Reply:
Ok, I get it, `np.min` will return `nan`

Reply:
Just to add some data to the discussion: @larsoner's approach is indeed faster:

```python
In [5]: %%timeit  
   ...: finite = np.isfinite(image) 
   ...: min_ = np.min(image[finite]) 
   ...: max_ = np.max(image[finite]) 
   ...:                                                                                             
25.6 ms ± 178 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [6]: %%timeit 
   ...: min_ = np.nanmin(image) 
   ...: max_ = np.nanmax(image) 
   ...:                                              
7.25 ms ± 43.6 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

I guess this is because boolean indexing performs a copy, regardless of the contents of the array (in this case, all the values are finite, non-nan).

nanmin is indeed slower than min:

```python
In [3]: %timeit np.min(image)                                                                                                 
2.7 ms ± 26.3 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [4]: %timeit np.nanmin(image)                                                                                              
3.69 ms ± 46.2 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

So, perhaps it should be a cascade:
- try min and max
- if they are not finite, try nanmin, nanmax
- if they are not finite, use isfinite

This would make the performance penalty be ~0 for finite arrays (which constitutes all inputs so far), and ~2.3x for finite + nan arrays, and ~∞ for arrays containing inf, which is somewhat appropriate. =P For me, this is preferable to adding a >30% penalty to all users by using nanmin/nanmax by default. As @djhoese points out, full performance can be restored for users with nans by passing the the clims up front.

Regarding @EtienneCmb's comment about what VisPy is, I would put it somewhere in the middle: for me, VisPy is a convenience layer for *fast* OpenGL-based visualisation. Usability is important, but speed even more so.